### PR TITLE
Migration to Python 3.4

### LINKED ISSUE #2: Make the project compatible with Python 3
- Ensure that Python 3 is installed in the system
- Install python3-pip
- Install python3 compatible libraries
- Ensure that the program works correctly with python3

### PR DESCRIPTION
**Pull request by gersilex**
Developed against the current major version of Python 3 shipped with Raspbian.

Install Script and main.py tested on common apt-based distributions:
- Debian 9.4 Docker Image
- Raspbian 8.0 on Raspberry Pi 2 (armv7l)
- Ubuntu 16.04.3 LTS

This was actually pretty easy. The code already was so clean and well-laid out so that there only was one tiny change in the `main.py`.

Closes #2

### CODE REVIEW

**Comment by bodiroga** [comment]
Hi @gersilex!

Can you please edit the ExecStart entry in the .service file (https://github.com/Tknika/serial2tcp-gateway/blob/master/systemd/serial2tcp-gateway.service) and also squash the 3 commits into a single one?

Many many thanks for your help!

File: systemd/serial2tcp-gateway.service [hunk 1]
```diff
@@ -9,7 +9,7 @@ User=serial2tcp-gateway
 Restart=always
 RestartSec=5
 
-ExecStart=/usr/bin/python /opt/serial2tcp-gateway/main.py
+ExecStart=/opt/serial2tcp-gateway/main.py
```
Comment (bodiroga) [inline systemd/serial2tcp-gateway.service:12]: Is this the best option or, following the syntax used previously, "/usr/bin/python3 /opt/serial2tcp-gateway/main.py"?

Comment (gersilex) [reply systemd/serial2tcp-gateway.service:12]: Yes it is. We should leave the interpreter to be started up to the script shebang (first line, starting with `#!`). The Shebang in the script in turn leaves it up to the system to choose the right interpreter for `python3`.

Users without root access and Mac OS X users often use those and we should not dictate where the python binary should be located on other peoples' systems. :)

Comment (bodiroga) [reply systemd/serial2tcp-gateway.service:12]: Perfect! Thank you very much also for your advice! :wink:

Comment (gersilex) [reply systemd/serial2tcp-gateway.service:12]: Anytime! I'm always happy if I can tell someone something new :smiley:

**Pull request merged**